eng/common/scripts/TypeSpec-Project-Process.ps1

```diff
@@ -165,15 +165,41 @@ if (Test-Path $tmpTspConfigPath) {
 
 $sdkProjectFolder = ""
 if ($generateFromLocalTypeSpec) {
+  Write-Host "Generating sdk code based on local type specs at $specRepoRoot."
   $sdkProjectFolder = Get-TspLocationFolder $tspConfigYaml $sdkRepoRootPath
   $tspLocationYamlPath = Join-Path $sdkProjectFolder "tsp-location.yaml"
   if (!(Test-Path -Path $tspLocationYamlPath)) {
-    Write-Error "Failed to find tsp-location.yaml in '$sdkProjectFolder', please make sure to provide CommitHash and RepoUrl parameters along with the local path of tspconfig.yaml in order to create tsp-location.yaml."
-    exit 1
+    # try to create tsp-location.yaml using HEAD commit of the local spec repo
+    Write-Warning "Failed to find tsp-location.yaml in '$sdkProjectFolder'. Trying to create tsp-location.yaml using HEAD commit of the local spec repo then proceed the sdk generation based upon local typespecs at $specRepoRoot. Alternatively, please make sure to provide CommitHash and RepoUrl parameters when running this script."
+    # set default repo to Azure/azure-rest-api-specs
+    $repo = "Azure/azure-rest-api-specs"
+    try {
+      Push-Location $specRepoRoot
+      $CommitHash = $(git rev-parse HEAD)
+      $gitRemotes = (git remote -v)
+      foreach ($remote in $gitRemotes) {
+        Write-Host "Checking remote $remote"
+        if ($remote.StartsWith("origin")) {
+          if ($remote -match '(.*)?github.com:(?<repo>[^/]*/azure-rest-api-specs).git') {
```

Comment:
Should this also support the private repo?

Reply:
I also think the ".git" at the end is optional.

Reply:
updated